app/src/nxt/java/com/pileproject/drive/programming/visual/block/sequence/TurnLeftSecBlock.java

```diff
@@ -17,64 +17,49 @@
 package com.pileproject.drive.programming.visual.block.sequence;
 
 import android.content.Context;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.widget.TextView;
 
 import com.pileproject.drive.R;
 import com.pileproject.drive.execution.ExecutionCondition;
 import com.pileproject.drive.execution.MachineController;
 import com.pileproject.drive.execution.NxtController;
 import com.pileproject.drive.util.development.Unit;
+import com.pileproject.drive.util.math.Range;
 
-import java.util.Locale;
+import java.math.BigDecimal;
 
 /**
  * Turn left for a while
  *
  * @author yusaku
  * @version 1.0 7-July-2013
  */
-public class TurnLeftSecBlock extends SequenceBlockHasNumText {
+public class TurnLeftSecBlock extends SequenceBlockHasNumberText {
 
-    public TurnLeftSecBlock(Context context) {
-        super(context);
-        View layout = LayoutInflater.from(context).inflate(R.layout.block_turn_left_sec, this);
-        numText = (TextView) layout.findViewById(R.id.block_numText);
-    }
+    // TODO: set from preference
+    private static final Range<BigDecimal> range = Range.closed(BigDecimal.ZERO, new BigDecimal(3));
 
-    @Override
-    public int getNum() {
-        double raw = Double.parseDouble(numText.getText().toString());
-        return (int) (raw * 1000);
-    }
+    // TODO: set from preference
+    private static final int PRECISION = 3;
 
-    @Override
-    public void setNum(int num) {
-        double raw = num / 1000.0;
-        numText.setText(String.format(Locale.ENGLISH, "%.3f", raw));
-    }
-
-    @Override
-    public Integer[] getDigit() {
-        return new Integer[]{1, 3};
+    public TurnLeftSecBlock(Context context) {
+        super(context, R.layout.block_turn_left_sec, R.id.block_numText);
     }
 
     @Override
-    public double getMax() {
-        return 3.0;
+    public int getPrecision() {
+        return PRECISION;
     }
 
     @Override
-    public double getMin() {
-        return 0.0;
+    public Range<BigDecimal> getRange() {
+        return range;
     }
 
     @Override
     public int action(
```

Comment:
I know this is my fault but if you have time, please remove break lines... (this is not mandatory)